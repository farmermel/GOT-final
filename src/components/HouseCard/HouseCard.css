.house-card {
  border: 4px black solid;
  border-radius: 5px;
  margin: 10px;
  padding: 10px;
  width: 350px;
}