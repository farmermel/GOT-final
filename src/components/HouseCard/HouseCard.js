import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './HouseCard.css';

const HouseCard = ({ card, clicked }) => {
  return (
    <Link to={{pathname:`house/${card.id}`}}>
      <div className='house-card'>
        <h1>Name: {card.name}</h1>
        <p>founded: {card.founded || 'no data available'}</p>
        <p>seats: {card.seats}</p>
        <p>titles: {card.titles}</p>
        <p>coat of arms: {card.coatOfArms}</p>
        <p>ancestral weapons: {card.ancestralWeapons}</p>
        <p>words: {card.words || 'no data available'}</p>
        {
          clicked && <p>sworn members: {card.swornMembers}</p>
        }
      </div>
    </Link>
  );  
};

HouseCard.propTypes = {
  card: PropTypes.objectOf(PropTypes.string),
  clicked: PropTypes.bool.isRequired
};

export default HouseCard;