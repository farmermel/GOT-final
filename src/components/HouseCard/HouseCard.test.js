/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import HouseCard from './HouseCard';

describe('HouseCard', () => {
  const wrapper = shallow(<HouseCard card={{house: 'greyjoy'}} />)
  it('matches snapshot', () => {

  })
})