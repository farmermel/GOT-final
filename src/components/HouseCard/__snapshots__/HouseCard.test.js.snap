// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HouseCard matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HouseCard
    card={
        Object {
            "house": "greyjoy",
          }
    }
    clicked={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="house-card"
>
        <h1>
                Name: 
        </h1>
        <p>
                founded: 
                no data available
        </p>
        <p>
                seats: 
        </p>
        <p>
                titles: 
        </p>
        <p>
                coat of arms: 
        </p>
        <p>
                ancestral weapons: 
        </p>
        <p>
                words: 
                no data available
        </p>
</div>,
      "replace": false,
      "to": Object {
        "pathname": "house/undefined",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Name: 
</h1>,
          <p>
            founded: 
            no data available
</p>,
          <p>
            seats: 
</p>,
          <p>
            titles: 
</p>,
          <p>
            coat of arms: 
</p>,
          <p>
            ancestral weapons: 
</p>,
          <p>
            words: 
            no data available
</p>,
          false,
        ],
        "className": "house-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Name: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Name: ",
            undefined,
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "founded: ",
              "no data available",
            ],
          },
          "ref": null,
          "rendered": Array [
            "founded: ",
            "no data available",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "seats: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "seats: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "titles: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "titles: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "coat of arms: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "coat of arms: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "ancestral weapons: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "ancestral weapons: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "words: ",
              "no data available",
            ],
          },
          "ref": null,
          "rendered": Array [
            "words: ",
            "no data available",
          ],
          "type": "p",
        },
        false,
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="house-card"
>
          <h1>
                    Name: 
          </h1>
          <p>
                    founded: 
                    no data available
          </p>
          <p>
                    seats: 
          </p>
          <p>
                    titles: 
          </p>
          <p>
                    coat of arms: 
          </p>
          <p>
                    ancestral weapons: 
          </p>
          <p>
                    words: 
                    no data available
          </p>
</div>,
        "replace": false,
        "to": Object {
          "pathname": "house/undefined",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Name: 
</h1>,
            <p>
              founded: 
              no data available
</p>,
            <p>
              seats: 
</p>,
            <p>
              titles: 
</p>,
            <p>
              coat of arms: 
</p>,
            <p>
              ancestral weapons: 
</p>,
            <p>
              words: 
              no data available
</p>,
            false,
          ],
          "className": "house-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Name: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Name: ",
              undefined,
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "founded: ",
                "no data available",
              ],
            },
            "ref": null,
            "rendered": Array [
              "founded: ",
              "no data available",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "seats: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "seats: ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "titles: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "titles: ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "coat of arms: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "coat of arms: ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "ancestral weapons: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "ancestral weapons: ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "words: ",
                "no data available",
              ],
            },
            "ref": null,
            "rendered": Array [
              "words: ",
              "no data available",
            ],
            "type": "p",
          },
          false,
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
